File: src/main/java/net/gegy1000/terrarium/server/map/source/raster/RasterDataAccess.java
package net.gegy1000.terrarium.server.map.source.raster;

import net.gegy1000.terrarium.server.map.source.tiled.TiledDataAccess;

public interface RasterDataAccess<T> extends TiledDataAccess {
    int getWidth();

    int getHeight();

    T get(int x, int y);
}

File: src/main/kotlin/net/gegy1000/terrarium/server/map/source/raster/RasterDataAccess.kt
package net.gegy1000.terrarium.server.map.source.raster

import net.gegy1000.terrarium.server.map.source.tiled.TiledDataAccess

interface RasterDataAccess<out T> : TiledDataAccess {
    val width: Int
    val height: Int

    fun get(x: Int, y: Int): T
}
